### ISSUE TITLE
Testing issue: tasmanian@7.7%gcc11.1.0

### ISSUE THREAD

**Issue by tldahlgren** [LLNL/spack#33119]
### Steps to reproduce the failure(s) or link(s) to test output(s)

@mkstoyanov 

Spack PR testing of the E4S stack has detected a failure.  More information is at:

- Build:  https://cdash.spack.io/build/1816824
- Stand-alone tests:  https://cdash.spack.io/viewTest.php?buildid=1816824

### Error message

Generate the Makefile:

```
-- The CXX compiler identification is GNU 9.4.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Tasmanian post-installation testing
-- Found MPI_CXX: /home/software/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spa/linux-ubuntu20.04-x86_64/gcc-11.1.0/mpich-4.0.2-zg3d5l3s6z6a3gth3cyr5h3qnuztuvz4/lib/libmpicxx.so (found version "4.0")
-- Found MPI: TRUE (found version "4.0")
-- Configuring incomplete, errors occurred!
See also "/tmp/root/spack-stage/spack-standalone-tests/yaz67t4hfsssxdswvlt7l3cvmjemsoua/tasmanian-7.7-rfpaeeq/CMakeFiles/CMakeOutput.log".
CMake Error at /home/software/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spa/linux-ubuntu20.04-x86_64/gcc-11.1.0/cmake-3.24.2-fzllxrl7cwsyvq7aob7yrxqvj6rj2qr6/share/cmake-3.24/Modules/CMakeFindDependencyMacro.cmake:47 (find_package):
By not providing "FindAMDDeviceLibs.cmake" in CMAKE_MODULE_PATH this
project has asked CMake to find a package configuration file provided by
"AMDDeviceLibs", but CMake did not find one.

Could not find a package configuration file provided by "AMDDeviceLibs"
with any of the following names:

AMDDeviceLibsConfig.cmake
amddevicelibs-config.cmake

Add the installation prefix of "AMDDeviceLibs" to CMAKE_PREFIX_PATH or set
"AMDDeviceLibs_DIR" to a directory containing one of the above files.  If
"AMDDeviceLibs" provides a separate development package or SDK, be sure it
has been installed.
Call Stack (most recent call first):
/home/software/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spa/linux-ubuntu20.04-x86_64/gcc-11.1.0/hip-5.2.3-vn2vm3lp5ziydgmhecsu3hn2lfvuywhq/lib/cmake/hip/hip-config.cmake:189 (find_dependency)
/home/software/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spa/morepadding/linux-ubuntu20.04-x86_64/gcc-11.1.0/tasmanian-7.7-rfpaeeqktnk4czjrgqxugi3jwjbrvtvk/lib/Tasmanian/TasmanianConfig.cmake:113 (find_package)
CMakeLists.txt:9 (find_package)
...
```

### Information on your system or the test runner

Runner: large-x86-pub-gitlab-runner-666b7757d9-5pfkj
OS Release: 5.4.144-69.257.amzn2.x86_64 
Spack version: spack-0.19.0.dev0 (10720fbb4e277c97a262dbf8d51f6b93c27dab2a) 
Repository revision: [10720fbb4e277c97a262dbf8d51f6b93c27dab2a](https://github.com/spack/spack/commit/10720fbb4e277c97a262dbf8d51f6b93c27dab2a)

### Additional information

_Update_: The E4S CI Stack (https://github.com/spack/spack/blob/develop/share/spack/gitlab/cloud_pipelines/stacks/e4s/spack.yaml) is the only one running the tests ATM  and it defines the image (ecpe4s/ubuntu20.04-runner-x86_64:2022-10-01).

### General information

- [X] I have reported the version of Spack/Python/Platform/Runner
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded any available logs
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by mkstoyanov** [comment]
CMake cannot find the AMD packages in the post-install tests.

I will investigate next week, but I'm on travel until Thursday.

Thanks for reporting this,

Miro
________________________________
From: Tamara Dahlgren ***@***.***>
Sent: Friday, October 7, 2022 4:02 PM
To: spack/spack ***@***.***>
Cc: Stoyanov, Miroslav ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] [spack/spack] Testing issue: ***@***.***%gcc11.1.0 (Issue #33119)


Steps to reproduce the failure(s) or link(s) to test output(s)

@mkstoyanov

Spack PR testing of the E4S stack has detected a failure. More information is at:

  *   Build: hxxps://cdash.spack.io/build/1816824
  *   Stand-alone tests: hxxps://cdash.spack.io/viewTest.php?buildid=1816824

Error message

Generate the Makefile:

-- The CXX compiler identification is GNU 9.4.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Tasmanian post-installation testing
-- Found MPI_CXX: /home/software/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spa/linux-ubuntu20.04-x86_64/gcc-11.1.0/mpich-4.0.2-zg3d5l3s6z6a3gth3cyr5h3qnuztuvz4/lib/libmpicxx.so (found version "4.0")
-- Found MPI: TRUE (found version "4.0")
-- Configuring incomplete, errors occurred!
See also "/tmp/root/spack-stage/spack-standalone-tests/yaz67t4hfsssxdswvlt7l3cvmjemsoua/tasmanian-7.7-rfpaeeq/CMakeFiles/CMakeOutput.log".
CMake Error at /home/software/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spa/linux-ubuntu20.04-x86_64/gcc-11.1.0/cmake-3.24.2-fzllxrl7cwsyvq7aob7yrxqvj6rj2qr6/share/cmake-3.24/Modules/CMakeFindDependencyMacro.cmake:47 (find_package):
By not providing "FindAMDDeviceLibs.cmake" in CMAKE_MODULE_PATH this
project has asked CMake to find a package configuration file provided by
"AMDDeviceLibs", but CMake did not find one.

Could not find a package configuration file provided by "AMDDeviceLibs"
with any of the following names:

AMDDeviceLibsConfig.cmake
amddevicelibs-config.cmake

Add the installation prefix of "AMDDeviceLibs" to CMAKE_PREFIX_PATH or set
"AMDDeviceLibs_DIR" to a directory containing one of the above files.  If
"AMDDeviceLibs" provides a separate development package or SDK, be sure it
has been installed.
Call Stack (most recent call first):
/home/software/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spa/linux-ubuntu20.04-x86_64/gcc-11.1.0/hip-5.2.3-vn2vm3lp5ziydgmhecsu3hn2lfvuywhq/lib/cmake/hip/hip-config.cmake:189 (find_dependency)
/home/software/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spa/morepadding/linux-ubuntu20.04-x86_64/gcc-11.1.0/tasmanian-7.7-rfpaeeqktnk4czjrgqxugi3jwjbrvtvk/lib/Tasmanian/TasmanianConfig.cmake:113 (find_package)
CMakeLists.txt:9 (find_package)
...


Information on your system or the test runner

Runner: large-x86-pub-gitlab-runner-666b7757d9-5pfkj
OS Release: 5.4.144-69.257.amzn2.x86_64
Spack version: spack-0.19.0.dev0 (10720fb)
Repository revision: 10720fbb4e277c97a262dbf8d51f6b93c27dab2a

Additional information

No response

General information

  *    I have reported the version of Spack/Python/Platform/Runner
  *    I have run spack maintainers <name-of-the-package> and @mentioned any maintainers
  *    I have uploaded any available logs
  *    I have searched the issues of this repo and believe this is not a duplicate

**Comment by mkstoyanov** [comment]
I managed to reproduce. Surprisingly, `spack test run tasmanian` fails but running the test manually after `spack load tasmanian` works just fine.

**Comment by mkstoyanov** [comment]
This should be resolved with #33381

Merged, closing the issue.
